File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& num, int target) {
        int o = 0;
        int p = num.size();
        for(int i =0;i<num.size();i++){
            if(num[o] + num[p-1] == target){
                return {o+1,p};
            }
            else if(num[o] + num[p-1] > target){
                p--;
            }
            else if(num[0] + num[p-1] < target){
                o++;
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        current_path = []
        nums.sort()
        def backtrack(i,curr_sum):
            if curr_sum == target:
                res.append(current_path.copy())
                return 
            if curr_sum > target:
                return

            for j in range(i, len(nums)):
                if j > i and nums[j] == nums[j - 1]:
                    continue
                current_path.append(nums[j])
                backtrack(j, curr_sum + nums[j])
                current_path.pop()
        backtrack(0,0)
        return res